scriptworker/gpg.py

```diff
@@ -1343,125 +1374,164 @@ def verify_signed_git_commit_output(output):
     finally:
-        rm(lockfile)
-    return basedir
+        rm_lockfile(context)
+
 
+def get_tmp_base_gpg_home_dir(context):
+    """Return the base_gpg_home_dir with a .tmp at the end.
 
-async def rebuild_gpg_homedirs_loop(context, basedir):
-    """Run a loop to update the git repo and rebuild the homedirs.
+    This function is really only here so we don't have to duplicate this
+    logic.
 
     Args:
         context (scriptworker.context.Context): the scriptworker context.
-        basedir (str): the base directory path, to create the gpg homedirs under.
-            This should be a temporary directory, so we can create the gpg
-            homedirs in the background without affecting anything.
-            The presence of this directory with a lockfile means something is
-            still generating the gpg homedirs.  The presence of this directory
-            without the lockfile means we're waiting to overwrite our gpg homedirs
-            with the contents of this tmp dir.
+
+    Returns:
+        str: the base_gpg_home_dir with .tmp at the end.
     """
-    if not context.config['sign_chain_of_trust'] and not context.config['verify_chain_of_trust']:
-        log.warning("sign_chain_of_trust and verify_chain_of_trust are False; exiting rebuild_gpg_homedirs_loop!")
-        return
-    while True:
-        await asyncio.sleep(context.config['poll_git_interval'])
-        if os.path.exists(basedir):
-            log.warning("rebuild_gpg_homedirs_loop: {} exists; sleeping another {}!".format(basedir, str(context.config['poll_git_interval'])))
-            continue
-        try:
-            message = "build_gpg_homedirs_from_repo error updating git repo!\n{}"
-            result = await retry_async(
-                update_signed_git_repo, retry_exceptions=(ScriptWorkerRetryException, ),
-                args=(context, )
-            )
-            if not result:  # no updates
-                continue
-            message = "build_gpg_homedirs_from_repo error in build_gpg_homedirs_from_repo!\n{}"
-            build_gpg_homedirs_from_repo(context, basedir=basedir)
-        except ScriptWorkerException as exc:
-            # We don't want these exceptions to be fatal; this will happen in the background
-            # of scriptworker tasks
-            log.error(message.format(str(exc)))
-            continue
+    return '{}.tmp'.format(context.config['base_gpg_home_dir'])
 
 
-def create_initial_gpg_homedirs():
-    """Create the initial gpg homedirs.
+def check_lockfile(context, name):
+    """Check for the lockfile.
+
+    Args:
+        context (scriptworker.context.Context): the scriptworker context
+        name (str): the name of the calling function
+
+    Returns:
+        bool: True if lockfile exists.
+    """
+    lockfile = context.config['gpg_lockfile']
+    if os.path.exists(lockfile):
+        log.warning("Skipping {}: lockfile {} exists!".format(name, lockfile))
+        return True
+
```

Comment:
Nit: `False` doesn't seem returned even though the documentation reads a bool is returned